extensions/cdc-postgres/src/test/java/com/hazelcast/jet/cdc/postgres/MultiTableCacheIntegrationTest.java

```diff
@@ -0,0 +1,314 @@
+                        new Order(10003, new Date(1455840000000L), 1002, 2, 106)),
+                new OrdersOfCustomer(
+                        new Customer(1003, "Edward", "Walker", "ed@walker.com"),
+                        new Order(10004, new Date(1456012800000L), 1003, REPEATS, 107)),
+                new OrdersOfCustomer(
+                        new Customer(1004, "Anne" + REPEATS, "Kretchmar", "annek@noanswer.org")));
+        expected.put(1005, new OrdersOfCustomer());
+        assertEqualsEventually(() -> getIMapContent(jet, CACHE), expected);
+    }
+
+    private StreamStage<ChangeRecord> fixOrdering(StreamStage<ChangeRecord> input) {
+        return input
+                .groupingKey(ChangeRecord::key)
+                .mapStateful(
+                        TimeUnit.SECONDS.toMillis(10),
+                        () -> new Sequence(0, 0),
+                        (lastSequence, key, record) -> {
+                            ChangeRecordImpl recordImpl = (ChangeRecordImpl) record;
+                            long source = recordImpl.getSequenceSource();
+                            long sequence = recordImpl.getSequenceValue();
+                            if (lastSequence.update(source, sequence)) {
+                                return record;
+                            }
+                            return null;
+                        },
+                        (TriFunction<Sequence, RecordPart, Long, ChangeRecord>) (sequence, recordPart, aLong) -> null);
+    }
+
+    @NotNull
+    private static Map<Integer, OrdersOfCustomer> getIMapContent(JetInstance jet, String name) {
+        return jet.<Integer, OrdersOfCustomer>getMap(name).entrySet().stream()
+                .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+    }
+
+    @NotNull
+    private static Map<Integer, OrdersOfCustomer> toMap(OrdersOfCustomer... ordersOfCustomers) {
+        return Arrays.stream(ordersOfCustomers).collect(Collectors.toMap(
+                orders -> orders.getCustomer().getId(), Function.identity()));
+    }
+
+    private static class Sequence {
+
+        private long source;
+        private long sequence;
+
+        Sequence(long source, long sequence) {
+            this.source = source;
+            this.sequence = sequence;
+        }
+
+        boolean update(long source, long sequence) {
+            if (this.source != source) { //sequence source changed for key
+                this.source = source;
+                this.sequence = sequence;
+                return true;
+            }
+
+            if (this.sequence < sequence) { //sequence is newer than previous for key
+                this.sequence = sequence;
+                return true;
+            }
+
+            return false;
+        }
+
+        @Override
+        public String toString() {
+            return "source=" + source + ", sequence=" + sequence;
+        }
+    }
+
+    private static class OrdersOfCustomer implements Serializable {
```

Comment:
I hoped to have 

```
class Customer {
  Set<Order> orders;
}
```

is there anything blocking us from having this? The `OrdersOfCustomer` is kind of equivalent, but not as nice.

Reply:
We can change it. The thing is though, that: 1) neither the current nor the suggested solution will actually work in a real cluster, due to IMDG not knowing how to deserialize these classes 2) the purpose of this test is to prove that we have SOME way of doing this join correctly and getting the data into IMDG. So... beautifying it should happen probably some place else.